TESTING.md

```diff
@@ -0,0 +1,59 @@
+# Testing CP-Ansible
+
+Each role has Molecule configuration which enable fast and easy development and testing with Docker containers.
+
+## Prerequisites
+
+Make sure Docker is running and increase the CPU/Memory available to Docker in your Preferences. Next install these python libraries:
+```
+pip install molecule docker
+```
+
+## Using Molecule
+
+Molecule comes with a CLI. Below are instructions for a few of its commands:
+
+### Running a role
+
+Each role has a default scenario in it already so running a role locally is as easy as:
+```
+cd roles/<role-name>
+molecule converge
+```
+
+### SSHing into a container
+
+Each docker container is named inside the molecule.yml file, copy the name and run:
+```
+molecule login -h <container-name>
+```
+
+### Running Tests
+
+To run the test cases, which are defined in the verify.yml playbooks, run:
+```
+molecule verify
+```
+
+### Destroying the Containers
+
+Simply run:
+```
+molecule destroy
+```
+
+## Creating Scenarios
+
+In some roles within the molecule directory you will find multiple subdirectories, which can be referred to as scenarios. Use a scenario as an inventory file, you can configure docker hosts, ansible groups, variables, etc. When developing new features for the role you can create a new scenario simply by duplicating the default one and customizing. Edit the verify.yml to have test assertions for your scenario.
```

Comment:
I would re-write this to something like:  Molecule uses the concept of scenarios to describe test cases.  A scenario consists of an inventory file, where you can define docker hosts, ansible groups, variables, etc.  

When developing new features for a role within CP-Ansible, we recommend duplicating the default scenario, customizing it to reflect the new functionality being added, and editing the `verify.yml` to apply your test assertions.

Further details can be found here: https://molecule.readthedocs.io/en/latest/getting-started.html#molecule-scenarios